core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java

```
@@ -848,15 +855,83 @@ public Graph build() {
+         * @param suffixFromProperties the suffix from property
+         * @see NamedOperationResolver#NamedOperationResolver(String)
+         * @see NamedViewResolver#NamedViewResolver(String)
+         */
+        private void validateGetFromCacheHook(final GraphConfig config, final Class<? extends Operation> operationClass, final Class<? extends GetFromCacheHook> hookClass, final String suffixFromProperties) {
+            if (store.isSupported(operationClass)) {
+                //Get Handler
+                final OperationHandler<Operation> addToCacheHandler = store.getOperationHandler(operationClass);
+                //If Handler is a AddToCacheHandler
+                final String suffix;
+                if (AddToCacheHandler.class.isAssignableFrom(addToCacheHandler.getClass())) {
+                    // get Suffix
+                    suffix = ((AddToCacheHandler<?>) addToCacheHandler).getSuffixCacheName();
+                } else {
+                    // otherwise get from properties
+                    LOGGER.warn(String.format(HANDLER_WAS_NOT_EXPECTED_TYPE_ADD_TO_CACHE_HANDLER, operationClass, AddToCacheHandler.class.getSimpleName(), suffixFromProperties));
+                    suffix = suffixFromProperties;
+                }
+                final List<GraphHook> hooks = config.getHooks();
+                //Is GetFromCacheHook missing
+                if (!hasHook(hooks, hookClass)) {
+                    //provide info about the graph not having the required hook
+                    LOGGER.info(HANDLER_WAS_SUPPLIED_BUT_WITHOUT_A_ADDING_WITH_SUFFIX, config.getGraphId(), operationClass, hookClass.getSimpleName(), hookClass.getSimpleName(), suffix);
+                    try {
+                        hooks.add(0, hookClass.getDeclaredConstructor(String.class).newInstance(suffix));
+                    } catch (final Exception e) {
+                        throw new RuntimeException(e);
```

Does this not just mask the exception type?

---

No, this will convert all checked exception into [unchecked exceptions](https://docs.oracle.com/javase/tutorial/essential/exceptions/runtime.html).

---

So would it be better instead to add a `throws` at the start if the method?

---

It would need to be caught somewhere else. I imagine that's not practical which is why it has been done like this.